sdk/core/core/inc/az_contract.h

```diff
@@ -10,13 +10,18 @@
 
 #include <_az_cfg_prefix.h>
 
-#define AZ_CONTRACT_ARG_NOT_NULL(arg) \
+#define AZ_CONTRACT(condition, error) \
   do { \
-    if ((arg) == NULL) { \
+    if (!(condition)) { \
       return AZ_ERROR_ARG; \
     } \
   } while (0)
 
+#define AZ_CONTRACT_ARG_NOT_NULL(arg) AZ_CONTRACT((arg) != NULL, AZ_ERROR_ARG)
+
+#define AZ_CONTRACT_ARG_VALID_SPAN(span) AZ_CONTRACT(az_span_is_valid(span), AZ_ERROR_ARG)
```

Comment:
ditto

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/135